frontend/src/composables/useShootEditor/helper.js

```diff
@@ -151,106 +159,80 @@ export class EditorCompletions {
-      let text = generateCompletionText(propertyName, completion.type, token.type)
-      if (token.type === 'firstArrayItem') {
-        text = '- ' + text
-      }
+      const text = generateCompletionText(propertyName, completion.type, token.type)
       const string = propertyName.toLowerCase()
-      const type = generateTypeText(completion.type, completion.format)
+      const typeText = generateTypeText(completion.type, completion.format)
 
       completionArray.push({
-        text,
-        string,
+        label: string,
+        displayLabel: text,
+        detail: typeText,
+        type: 'keyword', // TODO check weather we want to show diffrent icons
+        info: completion.description,
+        apply: text,
         property: propertyName,
-        type,
-        description: completion.description,
-        render (el, self, data) {
-          const document = el.ownerDocument
-          const { property, type, description } = data
-
-          const propertyElement = document.createElement('span')
-          propertyElement.className = 'property'
-          propertyElement.textContent = property
-
-          const typeElement = document.createElement('span')
-          typeElement.className = 'type'
-          typeElement.textContent = type
-
-          const propertyWrapper = document.createElement('div')
-          propertyWrapper.className = 'ghint-type'
-          propertyWrapper.appendChild(propertyElement)
-          propertyWrapper.appendChild(typeElement)
-
-          const descriptionElement = document.createElement('span')
-          descriptionElement.className = 'description'
-          descriptionElement.textContent = description
-
-          const descriptionWrapper = document.createElement('div')
-          descriptionWrapper.className = 'ghint-desc'
-          descriptionWrapper.appendChild(descriptionElement)
-
-          el.appendChild(propertyWrapper)
-          el.appendChild(descriptionWrapper)
-        },
       })
     })
+
     if (trim(token.string).length > 0) {
       completionArray = filter(completionArray, completion => {
-        if (exactMatch) {
-          return isEqual(completion.string, token.string)
+        let tokenString = token.string
+        if (token.type === 'firstArrayItem') {
+          tokenString = tokenString.replace(/.*\s-\s/, '')
         }
-        if (includes(words(token.string), completion.string)) {
-          // filter completion if already in this line - avoid duplicate completions
-          return false
+        if (exactMatch) {
+          return isEqual(completion.label, tokenString)
         }
         // filter completions that to not match text already in line
-        return includes(completion.string, token.string)
+        return includes(completion.label, tokenString)
       })
     }
+
     return completionArray
   }
 
   // get token at cursor position in editor with yaml content
-  _getYamlToken (cm, cur) {
+  _getYamlToken (cmView, cur) {
```

Comment:
These methods starting with underscore should be private and the underscore removed.